tests/rules/use-layers.test.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * @fileoverview Tests for use-layers rule.
+ * @author Nicholas C. Zakas
+ */
+
+//------------------------------------------------------------------------------
+// Imports
+//------------------------------------------------------------------------------
+
+import rule from "../../src/rules/use-layers.js";
+import css from "../../src/index.js";
+import { RuleTester } from "eslint";
+import dedent from "dedent";
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+	plugins: {
+		css,
+	},
+	language: "css/css",
+});
+
+ruleTester.run("use-layers", rule, {
+	valid: [
+		"@layer bar { a { color: red; } }",
+		"@layer foo { a { color: red; } }",
+		"@import 'foo.css' layer(foo);",
+		dedent`
+			@media (min-width: 600px) {
+				@layer foo {
+					a { color: bar }
+				}
+			}
+		`,
+		dedent`
+			@media (min-width: 600px) {
+				@layer foo {
+					a { color: bar }
+					@layer bar {
+						b { color: red; }
+					}
+				}
+			}
+		`,
+		{
+			code: "@layer { a { color: red; } }",
+			options: [{ allowUnnamedLayers: true }],
+		},
```

Comment:
Worth adding tests for importing into unnamed layers? e.g.

```css
@import 'foo.css' layer;
@import 'foo.css' layer();
```

Reply:
For the first one, yes. As noted in my other comment, `layer()` is actually a syntax error.